src/app/components/ServiceWorkerToaster/index.tsx

```diff
@@ -3,58 +3,112 @@
-              waitingWorker &&
-                waitingWorker.postMessage({ type: 'SKIP_WAITING' });
-              window.location.reload();
-            },
-            text: t(translations.serviceWorkerToaster.button),
-          },
-          timeout: 0,
-        },
-        'app-updated',
-      );
-    },
-    [t],
-  );
+  const fetchSw = useCallback(first => {
+    axios
+      .get(swUrl, {
+        headers: { 'Service-Worker': 'script' },
+      })
+      .then(({ data }) => {
+        if (!data) return;
+        const hash = soliditySha3(data);
+        if (first) setOldSW(hash || '');
+        else setNewSW(hash || '');
+      })
+      .catch(() => {});
+  }, []);
 
   useEffect(() => {
-    // If you want your app to work offline and load faster, you can change
-    // unregister() to register() below. Note this comes with some pitfalls.
-    // Learn more about service workers: https://bit.ly/CRA-PWA
+    if (!oldSW && newSW) setOldSW(newSW);
+    if (oldSW && newSW && oldSW !== newSW) setShow(true);
+
+    // eslint-disable-next-line
+  }, [oldSW, newSW]);
+
+  const updateSW = () => {
+    if (swRegistration) {
+      const waitingWorker = swRegistration && swRegistration.waiting;
+      waitingWorker && waitingWorker.postMessage({ type: 'SKIP_WAITING' });
+    } else {
+      serviceWorker.unregister();
+    }
+    window.location.reload();
+  };
+
+  useEffect(() => {
+    fetchSw(true);
     serviceWorker.register({
-      onUpdate: registration => onUpdateNotification(registration),
+      onUpdate: registration => {
+        setSwRegistration(registration);
+        setShow(true);
+      },
     });
     // eslint-disable-next-line
   }, []);
+  useEffect(() => {
+    const intId = setInterval(() => fetchSw(false), CHECK_TIME);
+    return () => clearInterval(intId);
+    // eslint-disable-next-line
+  }, []);
 
-  return <></>;
+  return (
+    <Dialog isOpen={show} className={styles.dialog}>
+      <div className="tw-container tw-mx-auto tw-px-4 text-center">
+        <StyledLogo className="mx-auto tw-mb-6" src={logoSvg} />
+        <p className="tw-text-white tw-mb-6">
+          <Trans
+            i18nKey={translations.serviceWorkerToaster.title}
+            components={[<strong></strong>]}
+            values={{ buildId: commitHash.substr(0, 7) }}
+          />
+        </p>
+        <p className="tw-mb-6">{t(translations.serviceWorkerToaster.notice)}</p>
+        <div className="tw-flex tw-flex-col tw-items-center">
+          <Button
+            className="tw-mb-3"
+            text={t(translations.serviceWorkerToaster.updateBtn)}
+            onClick={() => updateSW()}
+          />
+          <Button
+            className={styles.close + ' text-gold bg-transparent'}
+            text={t(translations.serviceWorkerToaster.closeBtn)}
+            onClick={() => setShow(false)}
+          />
+        </div>
+      </div>
+    </Dialog>
+  );
 }
+
+const StyledLogo = styled.img.attrs(_ => ({
+  alt: 'logo',
```

Comment:
Just a dummy change to force redeployment.
```suggestion
  alt: 'Sovryn',
```